src/DataCollection.Shared/ViewModels/DestinationRelationshipViewModel.cs

```diff
@@ -21,67 +21,66 @@
 using Esri.ArcGISRuntime.OpenSourceApps.DataCollection.Shared.Models;
 using Esri.ArcGISRuntime.Mapping.Popups;
 using System;
-using System.Collections.ObjectModel;
 using System.Linq;
 using System.Threading.Tasks;
+using System.Collections.Generic;
+using System.Windows.Input;
+using Esri.ArcGISRuntime.OpenSourceApps.DataCollection.Shared.Commands;
+using System.Threading;
 
 namespace Esri.ArcGISRuntime.OpenSourceApps.DataCollection.Shared.ViewModels
 {
     public class DestinationRelationshipViewModel : FeatureViewModel
-    { 
+    {
+        // Global cache reduces repetitive queries when loading multiple results
+        private static Dictionary<ArcGISFeatureTable, IEnumerable<PopupManager>> CachedTableResults = new Dictionary<ArcGISFeatureTable, IEnumerable<PopupManager>>();
+        // Synchronizes access to the global cache when loading multiple results in parallel
+        private static SemaphoreSlim cacheMutex = new SemaphoreSlim(1, 1);
```

Comment:
Should these private variables have an underscore? Or does being static change that style pattern?